iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -770,7 +770,10 @@ PortManager::acquirePublisherPortData(const capro::ServiceDescription& service,
                                       const RunnableName_t& runnable [[gnu::unused]], // @todo #25 Fix introspection
                                       const PortConfigInfo& portConfigInfo) noexcept
 {
-    /// @todo #25 Add unique check for single producer type - see acquireSenderPortData()
+    if (hasDuplicatePublisher<iox::build::CommunicationPolicy>(service, processName))
+    {
+        return cxx::error<PortPoolError>(PortPoolError::UNIQUE_PUBLISHER_PORT_ALREADY_EXISTS);
```

Comment:
We should call the errorHandler here if we have 1 to m communication. Otherwise `hasDuplicatePublisher` has unforseen side effects.

Reply:
That makes sense. However, I had the issue that some of the infos that are printed are only available when looping over publisher port list. I would remove the following:
````
<< publisherPortData->m_processName << "' with service '"
<< service.operator cxx::Serialization().toString() << "'.";
````
What do you think?

Reply:
Moved the `errorHandler` call outside and left the `LogWarn` inside

Reply:
This would be fine by me.